baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py

```diff
@@ -0,0 +1,189 @@
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+from baremetal_network_provisioning.common import constants as hp_const
+from baremetal_network_provisioning.ml2 import (hp_network_provisioning_driver
+                                                as np_drv)
+from baremetal_network_provisioning.ml2 import mechanism_hp as hp_mech
+
+import contextlib
+
+import mock
+from oslo_config import cfg
+
+# from neutron.extensions import portbindings
+from neutron.tests import base
+CONF = cfg.CONF
+
+
+class TestHPMechDriver(base.BaseTestCase):
+
+    def setUp(self):
+        super(TestHPMechDriver, self).setUp()
+        self.driver = hp_mech.HPMechanismDriver()
+        self.np_driver = np_drv.HPNetworkProvisioningDriver()
+        self.driver.initialize()
+        self.driver._load_drivers()
+
+    def _get_port_context(self, tenant_id, net_id, vm_id, network):
+        port = {'device_id': vm_id,
+                'device_owner': 'compute',
+                'binding:host_id': 'ubuntu1',
+                'name': 'test-port',
+                'tenant_id': tenant_id,
+                'id': 123456,
+                'network_id': net_id,
+                'binding:profile':
+                {'local_link_information': [{'switch_id': '11:22:33:44:55:66',
+                                             'port_id': 'Tengig0/1'}]},
+                'binding:vnic_type': 'baremetal',
+                'admin_state_up': True
+                }
+        return FakePortContext(port, port, network)
+
+    def _get_network_context(self, tenant_id, net_id, seg_id, shared):
+        network = {'id': net_id,
+                   'tenant_id': tenant_id,
+                   'name': 'test-net',
+                   'shared': shared}
+        network_segments = [{'segmentation_id': seg_id}]
+        return FakeNetworkContext(network, network_segments, network)
+
+    def _get_port_dict(self):
+        port_dict = {'port':
+                     {'segmentation_id': 1001,
+                      'access_type': hp_const.ACCESS,
+                      'switchports':
+                      [{'port_id': 'Tengig0/1',
+                          'switch_id': '11:22:33:44:55:66'}],
+                      'id': 123456,
+                      'is_lag': False}}
+        return port_dict
+
+    def test_create_port_precommit(self):
+        """Test create_port_precommit method."""
+        fake_port_dict = mock.Mock()
+        fake_context = mock.Mock()
+        with contextlib.nested(
+            mock.patch.object(hp_mech.HPMechanismDriver,
```

Comment:
why we are mocking _is_port_of_interest() is being mocked even though , you are passing valid port context.